bcda/api/requests.go

```diff
@@ -335,17 +350,32 @@ func (h *Handler) validateRequest(r *http.Request) ([]string, *fhirmodels.Operat
 	return resourceTypes, nil
 }
 
-func check429(jobs []models.Job, types []string, w http.ResponseWriter) ([]string, bool) {
+type duplicateTypeError struct{}
+
+func (e duplicateTypeError) Error() string {
+	return "Duplicate type found"
+}
+
+// check429 verifies that we do not have a duplicate resource type request.
+// Returns the unworkedTypes (if any)
+func check429(jobs []models.Job, types []string, version string) ([]string, error) {
 	var unworkedTypes []string
 
 	for _, t := range types {
 		worked := false
 		for _, job := range jobs {
 			req, err := url.Parse(job.RequestURL)
 			if err != nil {
-				log.Error(err)
```

Comment:
Since we weren't doing an early return here, line 351 would've panicked because req is nil in this case.

If we did an early return (when we only had boolean type), we would still write the 429 response AFTER we've written the OperationalOutcome.